### PR TITLE
BUG: Better float checking for FT

### LINKED ISSUE #8071: Issues loading fieldtrip data with MNE python
Sorry, I did not know where else I could ask this:
I am having challenges loading fieldtrip data with MNE python. When I enter the code below

```
import mne
original_data = mne.io.read_raw_fif('data.fif', preload=False)
original_info = original_data.info
data_from_ft = mne.read_epochs_fieldtrip('data.mat', original_info)
```
I get this error:

`TypeError: only size-1 arrays can be converted to Python scalars `

### PR DESCRIPTION
**Pull request by larsoner**
Closes #8071.

@acquayefrank can you see if this helps with your error at all?

### CODE REVIEW

File: mne/io/fieldtrip/utils.py [hunk 1]
```diff
@@ -203,12 +203,18 @@ def _set_sfreq(ft_struct):
...
         except KeyError:
             raise ValueError('No Source for sfreq found')
+        else:
+            t1, t2 = float(time[0]), float(time[1])
+            sfreq = 1 / abs(t1 - t2)
+    try:
+        sfreq = float(sfreq)
+    except TypeError:
+        warn('FieldTrip structure contained multiple sample rates, trying the '
+             f'first of:\n{sfreq}')
```
Comment (hoechenberger) [inline mne/io/fieldtrip/utils.py:216]:
```suggestion
             f'first of:\n{sfreq} Hz')
```

File: mne/io/fieldtrip/utils.py [hunk 2]
```diff
@@ -203,12 +203,18 @@ def _set_sfreq(ft_struct):
...
-            t1 = ft_struct['time'][0]
-            t2 = ft_struct['time'][1]
-            difference = abs(t1 - t2)
-            sfreq = 1 / difference
+            time = ft_struct['time']
         except KeyError:
             raise ValueError('No Source for sfreq found')
+        else:
+            t1, t2 = float(time[0]), float(time[1])
+            sfreq = 1 / abs(t1 - t2)
```
Comment (hoechenberger) [inline mne/io/fieldtrip/utils.py:211]: Why not simply
```suggestion
            sfreq = 1 / (t2 - t1)
```
?

Comment (larsoner) [reply mne/io/fieldtrip/utils.py:211]: Wanted to change the underlying code minimally, wasn't sure if there was a reason. But I agree `t2 - t1` is probably fine

**Comment by hoechenberger** [comment]
Just going ahead and merging – @acquayefrank if your problem is not fixed, please report back :)

**Comment by hoechenberger** [comment]
Thanks @larsoner

**Comment by acquayefrank** [comment]
@larsoner thanks for the help. But I still get the same error even though I'm using this version of the code.  cc @hoechenberger

**Comment by larsoner** [comment]
Can you paste the new traceback in #8071 ?

**Comment by acquayefrank** [comment]
Hello, @larsoner  I just did.

**Pull request merged**